.github/workflows/nightly-release-ui-test.yaml

```diff
@@ -0,0 +1,269 @@
+          ref: ${{ matrix.branch }}
+
+      # Not necessary to use yarn, but enables caching
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 14
+          cache: 'yarn'
+          cache-dependency-path: ./ui/yarn.lock
+
+      - name: Install
+        id: install
+        working-directory: ./ui
+        run: make deps
+
+      - name: Workspace Tests
+        id: workspace-test
+        working-directory: ./ui
+        run: make test-workspace
+
+      - name: Node Tests
+        id: node-test
+        working-directory: ./ui/packages/consul-ui
+        run: make test-node
+
+  frontend-build-oss:
+    runs-on: ubuntu-latest
+    needs: [frontend-cache]
+    strategy:
+      matrix:
+        branch: ${{ env.BRANCH_MATRIX }}
+    env:
+      JOBS: 2
+      CONSUL_NSPACES_ENABLED: 0
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          ref: ${{ matrix.branch }}
+
+      # Not necessary to use yarn, but enables caching
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 14
+          cache: 'yarn'
+          cache-dependency-path: ./ui/yarn.lock
+
+      - name: Install
+        id: install
+        working-directory: ./ui
+        run: make deps
+
+      - name: Ember Build OSS
+        id: build-oss
+        working-directory: ./ui/packages/consul-ui
+        run: make build-ci
+
+      - name: Upload OSS Frontend
+        uses: actions/upload-artifact@v3
+        with:
+          name: frontend-oss-${{ matrix.branch }}
+          path: .ui/packages/consul-ui/dist
+
+  frontend-test-oss:
+    runs-on: ubuntu-latest
+    needs: [frontend-build-oss]
+    strategy:
+      matrix:
+        branch: ${{ env.BRANCH_MATRIX }}
+        partion: ${{ env.EMBER_PARTITION_MATRIX }}
+    env:
+      CONSUL_NSPACES_ENABLED: 0
+      EMBER_TEST_REPORT: test-results/report-oss.xml #outputs test report for CircleCI test summary
+      EMBER_TEST_PARALLEL: true #enables test parallelization with ember-exam
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          ref: ${{ matrix.branch }}
+
+      # Not necessary to use yarn, but enables caching
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 14
+          cache: 'yarn'
+          cache-dependency-path: ./ui/yarn.lock
+
+      - name: Install
+        id: install
+        working-directory: ./ui
+        run: make deps
+
+      - name: Download OSS Frontend
+        uses: actions/upload-artifact@v3
+        with:
+          name: frontend-oss-${{ matrix.branch }}
+          path: .ui/packages/consul-ui/dist
+
+      - name: Ember Test OSS
+        id: cache
+        working-directory: ./ui/packages/consul-ui
+        run: node_modules/.bin/ember exam --split=$EMBER_PARTITION_TOTAL --partition=${{ matrix.partition }} --path dist --silent -r xunit
+
+      # TODO: add test-reporter: https://github.com/marketplace/actions/test-reporter
```

Comment:
Is this todo still relevant (if we already have the slack failure notification step)?

Reply:
Yes. `test-reporter` uploads the XML ember test results to Github so you can view them in the actions interface. Technically this is needed for parity with the Circle tests, but I didn't think it was needed for an MVP of nightly tests. Since it's hard to test these actions without merging them, I didn't want to have one more thing to debug 🤣 .